common/comic.cpp

```diff
@@ -340,6 +340,23 @@ QList<QString> Comic::findValidComicFilesInFolder(const QString &path)
     return validComicFiles;
 }
 
+void Comic::moveAndConnectToThread(QThread *thread)
+{
+    Q_ASSERT(thread);
+    moveToThread(thread);
+
+    void (Comic::*errorOpening0)() = &Comic::errorOpening;
+    void (Comic::*errorOpening1)(QString) = &Comic::errorOpening;
+
+    connect(this, errorOpening0, thread, &QThread::quit, Qt::QueuedConnection);
+    connect(this, errorOpening1, thread, &QThread::quit, Qt::QueuedConnection);
+    connect(this, &Comic::imagesLoaded, thread, &QThread::quit, Qt::QueuedConnection);
+    connect(this, &Comic::invalidated, thread, &QThread::quit, Qt::QueuedConnection);
```

Comment:
When I was implementing this function, I thought that with the default connection type [`Qt::AutoConnection`](https://doc.qt.io/qt-5/qt.html#ConnectionType-enum), the actual connection type is determined statically inside the `QObject::connect()` call. But after some debugging, reading Qt source code and rereading the documentation, I realized that this is a truly different connection type:
> The connection type is determined when the signal is emitted.

Since both `ComicController` classes connected to all signals using `Qt::AutoConnection`, calling this function instead changes their behavior. If this change is undesirable, I can restore the use of `Qt::AutoConnection` by passing one more argument to this function - `qthreadQuitConnectionType`.

Reply:
I think we don't want Qt::QueuedConnection in the server side. Do you use YACReader for iOS? Are these changes tested with it?

Reply:
No, I don't use or test the server. I have noticed the leak fixed in this PR while looking at code patterns similar to the one fixed in #203 and comparing very similar `Render` and `ComicController*` connection code. Then decided to unify the code to prevent future accidental divergence.

Reply:
Now I see that there is no real leak in the current implementation as Server never calls `Comic::invalidate()`. Though it might call it in the future and connecting to `Comic::invalidated` still make sense.

So it turns out that this pull request is pure code refactoring, not a fix.